src/main/java/org/xbill/DNS/NioTcpClient.java

```diff
@@ -44,6 +44,9 @@ private void processPendingRegistrations() {
         if (!state.channel.isConnected()) {
           state.channel.register(selector, SelectionKey.OP_CONNECT, state);
         } else {
+          if (state.channel.keyFor(selector) == null) {
```

Comment:
I encountered: 
```
Exception in thread "dnsjava NIO selector" java.lang.NullPointerException: Cannot invoke "java.nio.channels.SelectionKey.interestOps(int)" because the return value of "java.nio.channels.SocketChannel.keyFor(java.nio.channels.Selector)" is null
	at org.dnsjava/org.xbill.DNS.NioTcpClient.processPendingRegistrations(NioTcpClient.java:47)
	at org.dnsjava/org.xbill.DNS.NioClient.runTasks(NioClient.java:168)
	at org.dnsjava/org.xbill.DNS.NioClient.runSelector(NioClient.java:133)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Exception in thread "main" java.io.IOException: Timed out while trying to resolve google.com./A, id=25831
	at org.dnsjava/org.xbill.DNS.Resolver.send(Resolver.java:172)
	at org.dnsjava/org.xbill.DNS.tools.dig.main(dig.java:227)
Caused by: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at org.dnsjava/org.xbill.DNS.Resolver.send(Resolver.java:153)
	... 1 more
```

The merge with the outer condition didn't work out for me.